Libraries/JavaScriptAppEngine/Initialization/InitializeJavaScriptAppEngine.js

```diff
@@ -70,6 +70,15 @@ function polyfillGlobal(name, newValue, scope = GLOBAL) {
   Object.defineProperty(scope, name, {...descriptor, value: newValue});
 }
 
+/**
+ * Polyfill a module if it is not already defined in `scope`.
+ */
+function polyfillIfNeeded(name, polyfill, scope = GLOBAL) {
+  if (typeof scope[name] === 'undefined') {
+    Object.defineProperty(scope, name, {value: polyfill});
```

Comment:
I added a parameter to configure the descriptor and used it to make navigator enumerable/writable/configurable. For the Number polyfill the properties are all not enumerable/writable/configurable so that is fine.